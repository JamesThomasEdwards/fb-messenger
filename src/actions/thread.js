export const RECEIVE_THREADS = "RECEIVE_THREADS";

export const receiveThreads = threads => ({
  type: RECEIVE_THREADS,
  threads
});
