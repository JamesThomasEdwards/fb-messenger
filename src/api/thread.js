import throwRandomError from "./throwRandomError";

export const fetchThreads = () => {
  throwRandomError();
  return fetch("/mocks/threads.json", {
    method: "get"
  }).then(response => {
    try {
      throwRandomError();
      return response.json();
    } catch (e) {
      return { threads: "🐛🐛🐛🐛🐛" };
    }
  });
};
