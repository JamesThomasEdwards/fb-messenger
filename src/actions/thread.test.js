import { receiveThread, RECEIVE_THREAD } from "./thread";

describe("Thread action", () => {
  it(`should return a ${RECEIVE_THREAD} action`, () => {
    const thread = { id: "1" };

    expect(receiveThread(thread)).toMatchSnapshot();
  });
});
