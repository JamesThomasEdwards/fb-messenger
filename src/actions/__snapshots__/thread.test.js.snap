// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thread action should return a RECEIVE_THREAD action 1`] = `
Object {
  "thread": Object {
    "id": "1",
  },
  "type": "RECEIVE_THREAD",
}
`;
